source/simulator/initial_conditions.cc

```diff
@@ -214,6 +91,55 @@ namespace aspect
 
         std::vector<types::global_dof_index> local_dof_indices (finite_element.dofs_per_cell);
 
+#if DEAL_II_VERSION_GTE(8,5,0)
+        const VectorFunctionFromScalarFunctionObject<dim, double> &advf_init_function =
+          (advf.is_temperature()?
+           VectorFunctionFromScalarFunctionObject<dim, double>(std_cxx11::bind(&InitialConditions::Interface<dim>::initial_temperature,
+                                                                               std::ref(*initial_conditions),
+                                                                               std_cxx11::_1),
+                                                               introspection.component_indices.temperature,
+                                                               introspection.n_components):
+           VectorFunctionFromScalarFunctionObject<dim, double>(std_cxx11::bind(&CompositionalInitialConditions::Interface<dim>::initial_composition,
+                                                                               std::ref(*compositional_initial_conditions),
+                                                                               std_cxx11::_1,
+                                                                               n-1),
+                                                               introspection.component_indices.compositional_fields[n-1],
+                                                               introspection.n_components));
+
+        const ComponentMask advf_mask =
+          (advf.is_temperature()?
+           introspection.component_masks.temperature:
+           introspection.component_masks.compositional_fields[n-1]);
+
+        VectorTools::interpolate(*mapping,
+                                 dof_handler,
+                                 advf_init_function,
+                                 initial_solution,
+                                 advf_mask);
+
+        if (n == 1 && parameters.normalized_fields.size()>0)
+          for (typename DoFHandler<dim>::active_cell_iterator cell = dof_handler.begin_active();
+               cell != dof_handler.end(); ++cell)
+            if (cell->is_locally_owned())
+              {
+                fe_values.reinit (cell);
+
+                // Go through the support points for each dof
+                for (unsigned int i=0; i<finite_element.base_element(base_element).dofs_per_cell; ++i)
+                  {
+                    // if it is specified in the parameter file that the sum of all compositional fields
+                    // must not exceed one, this should be checked
+                    double sum = 0;
+                    for (unsigned int m=0; m<parameters.normalized_fields.size(); ++m)
+                      sum += compositional_initial_conditions->initial_composition(fe_values.quadrature_point(i),parameters.normalized_fields[m]);
+                    if (std::abs(sum) > 1.0+1e-6)
```

Comment:
And there is another occurence in the other code path, please replace that as well. I realize it was there before your change, but this is the opportunity to clean up :smile: